source/server/listener_manager_impl.cc

```diff
@@ -289,6 +304,16 @@ bool ListenerManagerImpl::addOrUpdateListener(const Json::Object& json) {
   return true;
 }
 
+bool ListenerManagerImpl::hasListenerWithAddress(const ListenerList& list,
+                                                 const Network::Address::Instance& address) {
+  for (const auto& listener : list) {
```

Comment:
We need to remove from the middle of the list, so list is probably better. Either way, this could almost never runs in the grand scheme of things so I don't think it matters.